[examples] Update guestbook troubleshooting guide w/ note about reset-by...

jayunit100 opened the pull request:
Clarify the common reason why we get "reset by peer" during guestbook startup

piosz committed:
Added new command clusterinfo to kubectl

This is first version of the command. It prints IPs of master and cluster
services. Should be improved once generalized labels are implemented #341.

It requires label kubernet.io/cluster-service=true set for cluster services.

Follow up cl after discussion in #4417.

ixdy committed:
Renable code coverage collection in unit tests, including on Travis.

When code coverage is not being collected, just issue a single 'go test'
command, as is already done. Go will internally parallize execution.

When code coverage is being collected, it is necessary to issue separate
'go test' commands for each package, since Go does not support
collecting coverage across packages. Using xargs -P will parallelize
these invocations, however, speeding up test execution.
The number of simultaneous processes to use can be specified with
KUBE_COVERPROCS.

Update Travis config to pass along the number of CPUs to use for running
tests.

Commit:
Rework definition of pods.

This rephrases and expands the definition of pods in an attempt to be:
a) much more explicit about what technically constitutes a pod;
b) agnostic towards particular container implementations.

A change along these lines will be critical as Kubernetes starts to
support alternative container runtimes (for example, #2725).

In trying to be as clear as possible, this turned into a larger rewrite
than I had originally anticipated, but I am fairly confident that it is
still effectively a no-op in terms of the current Kubernetes
implementation.  But I stand correctable :-).

The change also introduces some other minor cleanup and drops the
link to the `lmctfy` paper, which seemed like a tenuous reference
(perhaps there is something more specific within the lmctfy project we
could link to instead?)

Commit:
docs/pods: update to address comments

Commit:
Support other names for GNU tar
The build is only looking for GNU tar as gtar on OSX, which is the name used when installed using brew. Macports installs it as gnutar, so check for that name if gtar is not found.

smarterclayton committed:
Minimal status mutation change

    PUT /api/v1beta3/namespaces/default/pods/foo/status
    {
      "metadata": {...}, // allowed for valid values
      "spec": {}, // ignored
      "status": {...}, // allowed, except for Host
    }

Exposes the simplest possibly change. Needs a slight refactoring
to RESTUpdateStrategy to split merging which can be done in a
follow up.

sub-mod committed:
v1beta3 api json for grafana example

Commit:
Update file limits from testing

lavalamp committed:
clean up poller test to not make infinite log messages

lavalamp committed:
add occasional polling to reflector

dchen1107 committed:
Hide POD container from end users

resouer committed:
Fix broken validation tips and typo in guide

Fix:
1. validation tips broken the list in doc
2. fix typo in command line

derekwaynecarr committed:
Further vagrant reliability improvements.

bprashanth committed:
Retry resizing replication controllers in kubectl

vmarmol committed:
Updating github.com/stretchr/testify godep.

thockin committed:
Change the v1b3 default for service ContainerPort

In v1b1 and v1b2 we choose the "first defined port" if you do not specify a
ContainerPort.  I am proposing that v1b3 just assume the ContainerPort is the
same as the service port unless explicitly provided.  This leaves named ports
for now, but that is under discussion on its own.

This is strictly compatible, though to implement this we have to leave the
internal objects with the looser behavior until v1b[12] die.  This also adds a
link dependency so that when we DO kill v1b[12] the endpoints controller will
blow up, prompting a fix.

satnam6502 committed:
Update cluster level docs for proxy access

lavalamp committed:
Revert "Revert "Adding converter functions to convert field label selectors to internal version before matching""

This reverts commit 5f35a670029e3e94e7f0777be1668f4f62d4b522.

lavalamp committed:
fix e2e failure

cjcullen committed:
Fix leaking PDs from pd.go e2e test

Also upped the timeouts, because Slow Test > Flaky Test.

satnam6502 committed:
Fix URL for Elasticsearch

jayunit100 committed:
[examples] Update guestbook troubleshooting guide w/ note about reset-by-peer frontend startup message.

roberthbailey commented on examples/guestbook/README.md:
````diff
@@ -464,7 +464,7 @@ When you go to localhost:8000, you might not see the page at all.  Testing it wi
 ```
 
-This means the web frontend isn't up yet.  Wait a while, possibly about 2 minutes or more, depending on your set up. Also, run a *watch* on docker ps, to see if containers are cycling on and off or not starting.
+This means the web frontend isn't up yet.  So "reset by peer", is occuring because you are trying to access the right port, but nothing is bound to that port yet.  Wait a while, possibly about 2 minutes or more, depending on your set up. Also, run a *watch* on docker ps, to see if containers are cycling on and off or not starting.
````
Can you reword this slightly. I'd suggest:

This means the web frontend isn't up yet ("reset by peer" is occurring because you are trying to access the right port, but nothing is bound to that port yet).

brendandburns committed:
Kubernetes version v0.12.0

brendandburns committed:
Kubernetes version v0.12.0-dev

piosz committed:
Added cluster-service lavel to monitoring services

Monitoring services will be now displayed in kubectl cluster command (#4657).

This partially addresses #4620

gtaylor committed:
Correct typo in guestbook/README.md

fgrzadkowski committed:
Fix hack/test-cmd.sh so that it passes on a local machine

justinsb committed:
Use internal ip for connecting to salt master, with clearer env var

justinsb committed:
Make public-ip detection more robust for AWS, enable SSH function

justinsb committed:
Use new config file & sync with GCE script

mikedanese committed:
Only log on failing cases in auth integration test

vmarmol committed:
Updating github.com/docker/docker/pkg/term Godep.

This matches the version of the rest of the Docker package. There is no
actual code change, but the mistmatch made it impossible to udpate the
dependency (or any that depended on this).

derekwaynecarr committed:
nodecontroller sync cloud was not setting external id in all code paths

philips committed:
docs: fix ior to or

cjcullen committed:
Remove pd.sh now that we have pd.go

jayunit100 committed:
Guestbook examples: Quick note about v1beta1 and v1beta3 In the README

jayunit100 commented:
This is a minor change ; and  has gotten stale in my github fork ;  Im going to redo this pull request.  thanks for the comments ill implement them.
